gdpr/tests/test_gdpr_api.py

```diff
@@ -0,0 +1,227 @@
+import urllib.parse
+import uuid
+from typing import Optional
+
+import pytest
+import requests_mock
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+from django.utils import timezone
+from rest_framework.test import APIClient
+
+from children.factories import ChildWithGuardianFactory
+from events.factories import EnrolmentFactory, TicketSystemPasswordFactory
+from gdpr.consts import CLEARED_VALUE
+from gdpr.service import clear_data
+from gdpr.tests.conftest import get_api_token_for_user_with_scopes
+from subscriptions.factories import FreeSpotNotificationSubscriptionFactory
+from subscriptions.models import FreeSpotNotificationSubscription
+from users.factories import GuardianFactory, UserFactory
+from users.models import Guardian
+
+User = get_user_model()
+
+
+def request_gdpr_delete(
+    user,
+    id_value,
+    scopes=(settings.GDPR_API_DELETE_SCOPE,),
+    query_params=None,
+    data=None,
+):
+    gdpr_api_client = APIClient()
+
+    with requests_mock.Mocker() as req_mock:
+        auth_header = get_api_token_for_user_with_scopes(user, scopes, req_mock)
+        gdpr_api_client.credentials(HTTP_AUTHORIZATION=auth_header)
+
+        if query_params:
+            query = "?" + urllib.parse.urlencode(query_params)
+        else:
+            query = ""
+
+        request_kwargs = {"format": "json"}
+        if data:
+            request_kwargs["data"] = data
+
+        return gdpr_api_client.delete(
+            reverse(
+                "helsinki_gdpr:gdpr_v1",
+                kwargs={settings.GDPR_API_MODEL_LOOKUP: id_value},
+            )
+            + query,
+            **request_kwargs,
+        )
+
+
+def assert_clear(user, original_password: Optional[str] = None):
+    if original_password:
+        assert user.password != original_password
+    assert user.first_name == ""
+    assert user.last_name == ""
+    assert user.username
+    try:
+        guardian = user.guardian
+        child = guardian.children.first()
+        assert guardian.first_name == CLEARED_VALUE
+        assert guardian.last_name == CLEARED_VALUE
+        assert child.name == ""
+    except Guardian.DoesNotExist:
+        pass
```

Comment:
Shouldn't the user.email and guardian.email be checked to be cleared too here?

Reply:
Added